picasso/src/main/java/com/squareup/picasso3/Picasso.java

```diff
@@ -710,7 +736,7 @@ public Builder callFactory(@NonNull Call.Factory factory) {
     @NonNull
     public Builder executor(@NonNull ExecutorService executorService) {
       checkNotNull(executorService, "executorService == null");
-      if (this.service != null) {
+      if (!isChild && this.service != null) {
         throw new IllegalStateException("Executor service already set.");
```

Comment:
remove these checks and the isChild flag. the checks don't add any value.

Reply:
yea, I went back and forth on this.  I guess it made sense before, when passing Builders around?

Reply:
updated.